install/kubernetes/helm/subcharts/mixer/templates/config.yaml

```diff
@@ -350,29 +350,29 @@ metadata:
   name: requestcount
   namespace: {{ .Release.Namespace }}
   labels:
-    app: {{ template "mixer.name" . }}
-    chart: {{ template "mixer.chart" . }}
-    heritage: {{ .Release.Service }}
-    release: {{ .Release.Name }}
+    app.kubernetes.io/name: {{ template "mixer.name" . }}
+    helm.sh/chart: {{ template "mixer.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
 spec:
   value: "1"
   dimensions:
     reporter: conditional((context.reporter.kind | "inbound") == "outbound", "source", "destination")
     source_workload: source.workload.name | "unknown"
     source_workload_namespace: source.workload.namespace | "unknown"
     source_principal: source.principal | "unknown"
-    source_app: source.labels["app"] | "unknown"
-    source_version: source.labels["version"] | "unknown"
+    source_app: source.labels["app.kubernetes.io/name"] | source.labels["app"] | "unknown"
+    source_version: source.labels["app.kubernetes.io/version"] | source.labels["version"] | "unknown"
```

Comment:
Here's an example of where this PR doesn't follow the final recommandation of the github issue - see: https://github.com/istio/istio/issues/10724#issuecomment-452211765

For backward compat, check version first, then check for the new label.

Same for line 370 below - and other lines. I didn't look for all of them.

The same kind of change is needed for the "app" labels (check instance, then name, then the original "app" label - see my earlier comment on this).

Reply:
+1
John's feedback can probably help solve some of the unit test failures too..

Reply:
Thank you for pointing this out. I missed that